backend/src/database/queries.js

```diff
@@ -59,58 +59,154 @@ export const CREATE_TABLES_QUERIES = [
   CREATE_MENTIONS_TABLE_QUERY,
 ]
 
-export const INSERT_NEW_CARD_QUERY =
-  'INSERT INTO cards (id, title, description, createdAt, chatIds, users, author, project_id) VALUES (?, ?, ?, ?, ?, ?, ?, ?)'
+export const INSERT_NEW_CARD_QUERY = `
+  INSERT INTO cards (id, title, description, createdAt, chatIds, users, author, project_id)
+  VALUES (?, ?, ?, ?, ?, ?, ?, ?)
+`
+
+export const INSERT_TASK_QUERY = `
+  INSERT OR REPLACE INTO tasks (id, isDone, text, priority, status, author, users, card_id)
+  VALUES (?, ?, ?, ?, ?, ?, ?, ?)
+`
+
+export const INSERT_FILES_QUERY = `
+  INSERT OR REPLACE INTO files (id, type, url, card_id)
+  VALUES (?, ?, ?, ?)
+`
+
+export const INSERT_MENTION_QUERY = `
+  INSERT OR REPLACE INTO mentions (id, text, user_id)
+  VALUES (?, ?, ?)
+`
+
+export const INSERT_NEW_USERS_TO_CARD_QUERY = `
+  UPDATE cards SET users = ? WHERE id = ?
+`
+
+export const UPDATE_CARD_QUERY = `
+  UPDATE cards SET description = ?, updatedAt = ? WHERE id = ?
+`
+
+export const SELECT_USER_TASKS_QUERY = `
+  SELECT tasks.* FROM tasks
+  JOIN cards ON tasks.card_id = cards.id
+  WHERE cards.project_id = ? AND tasks.users LIKE ?
+`
+
+export const SELECT_ALL_CARDS_BY_PROJECTID_QUERY = (orderBy) => `
+  SELECT cards.*, 
+    COALESCE((
+      SELECT json_group_array(json_object('id', files.id, 'type', files.type, 'url', files.url))
+      FROM files WHERE cards.id = files.card_id), '[]') AS files 
+  FROM cards 
+  LEFT JOIN files ON cards.id = files.card_id 
+  WHERE cards.project_id = ? 
+    AND (cards.author = ? OR EXISTS (
+      SELECT 1 
+      FROM json_each(cards.users) 
+      WHERE json_each.value = ?
+    ))
+  GROUP BY cards.id 
+  ORDER BY ${orderBy} DESC
+`
+
+export const SELECT_ALL_CARDS_WITH_TASKS_BY_PROJECTID_QUERY = ` 
+  SELECT 
+    c.*, 
+    json_group_array(
+      json_object(
+        'id', t.id,
+        'isDone', t.isDone,
+        'text', t.text,
+        'users', t.users,
+        'author', t.author,
+        'priority', t.priority,
+        'status', t.status
+      )
+    ) AS tasks,
+    json_group_array(
+      json_object(
+        'id', f.id,
+        'type', f.type,
+        'url', f.url
+      )
+    ) AS files
+  FROM 
+    cards c
+  LEFT JOIN 
+    tasks t ON c.id = t.card_id
+  LEFT JOIN 
+    files f ON c.id = f.card_id
+  WHERE 
+    c.project_id = ?
+  GROUP BY 
+    c.id
+
+`

```

Comment:
Why do we actually have two different queries for one task? 🙂 The route `/cards` ([getAllProjectCards](https://github.com/prspktvs/dun/blob/d1791944525a763d1894a0e69e0d6319eb89e150/backend/src/api/cards.js#L43)) should return all available cards with files and tasks. (Now it only returns the cards with files)